src/picongpu/include/plugins/ChargeConservation.tpp

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright 2013-2015 Axel Huebl, Heiko Burau, Rene Widera, Felix Schmitt
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "math/vector/Int.hpp"
+#include "math/vector/Float.hpp"
+#include "math/vector/Size_t.hpp"
+#include "math/vector/math_functor/abs.hpp"
+#include "math/vector/math_functor/max.hpp"
+#include "cuSTL/container/PseudoBuffer.hpp"
+#include "dataManagement/DataConnector.hpp"
+#include "fields/FieldJ.hpp"
+#include "math/Vector.hpp"
+#include "cuSTL/algorithm/mpi/Gather.hpp"
+#include "cuSTL/algorithm/mpi/Reduce.hpp"
+#include "cuSTL/container/DeviceBuffer.hpp"
+#include "cuSTL/container/HostBuffer.hpp"
+#include "cuSTL/algorithm/kernel/Foreach.hpp"
+#include "cuSTL/algorithm/host/Foreach.hpp"
+#include "cuSTL/cursor/NestedCursor.hpp"
+#include "lambda/Expression.hpp"
+#include <sstream>
+#include "algorithms/ForEach.hpp"
+#include "cuSTL/algorithm/kernel/Reduce.hpp"
+#include "nvidia/functors/Add.hpp"
+
+namespace picongpu
+{
+
+ChargeConservation::ChargeConservation()
+    : name("ChargeConservation: Print the maximum charge deviation between particles and div E to textfile 'chargeConservation.dat'"),
+      prefix("chargeConservation"), output_file(NULL)
+{
+    Environment<>::get().PluginConnector().registerPlugin(this);
+}
+
+ChargeConservation::~ChargeConservation()
+{
+    delete this->output_file;
+    this->output_file = NULL;
+}
+
+void ChargeConservation::pluginRegisterHelp(po::options_description& desc)
+{
+    desc.add_options()
+        ((this->prefix + ".period").c_str(),
+        po::value<uint32_t > (&this->notifyPeriod)->default_value(0), "enable plugin [for each n-th step]");
+}
+
+std::string ChargeConservation::pluginGetName() const {return this->name;}
+
+void ChargeConservation::pluginLoad()
+{
+    Environment<>::get().PluginConnector().setNotificationPeriod(this, this->notifyPeriod);
+
+    this->output_file = new std::ofstream("chargeConservation.dat");
+    *this->output_file << "#timestep max-charge-deviation[unitless charge]" << std::endl;
```

Comment:
please write `unitless_charge` thus the column count in the geader is the same like in the data.
Can you please add a third column with the unit for charge than the file can used without information from other files.